src/app/domUtils.ts

```diff
@@ -97,7 +97,17 @@ const getScrollPadding = () => {
 };
 
 export const scrollTo = (elem: HTMLElement | Element | string) => {
-  return scrollToElement(elem, {offset: getScrollPadding()});
+  const scroll = () => scrollToElement(elem, {offset: getScrollPadding()});
+
+  document?.addEventListener(
+    'load', scroll, true
+  );
+
+  setTimeout(
+    () => document?.removeEventListener('load', scroll, true),
+    3000
+  );
+  return scroll();
```

Comment:
This looks like a double fire would occur under load conditions. Would it make sense to use a `document.readyState === 'complete'` check to split these? That would also let you skip the listeners entirely when the document is loaded.

Reply:
This is listening for images that lazy load, which happens as the page scrolls, throwing off the coordinates of the scroll target.
I can debounce the listener to avoid rapid multiple fire (in testing, I saw something like 97 image load events on a run), but I don't notice any janky behavior from the multiple calls.

Reply:
As an example, if you go [here](https://rex-web-production.herokuapp.com/books/microbiology/pages/8-3-cellular-respiration), search for "chemical reaction", and then select the result (in section 8.1) that begins "Overall," the target result will scroll a bit off the top of the screen.